server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStatsRequest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.remotestore.stats;
+
+import org.opensearch.action.support.broadcast.BroadcastRequest;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * Encapsulates all remote store stats
+ *
+ * @opensearch.internal
+ */
+public class RemoteStoreStatsRequest extends BroadcastRequest<RemoteStoreStatsRequest> {
+
+    private String[] shards = new String[0];
```

Comment:
shards = new String[0];
the above assignment is always replaced with the constructor value. we can have no assignment here, right?

Reply:
moved this to ctor